app/services/v1/question/update.rb

```diff
@@ -15,10 +15,16 @@ def call
     raise ActiveRecord::RecordNotSaved, I18n.t('question.error.not_uniq_variable') if new_variable_is_taken?(new_variables)
 
     previous_var = question_variables
+
+    changed_vars = changed_variables_preview(previous_var, question_params)
+    raise ActiveRecord::RecordNotSaved, I18n.t('question.error.formula_update_in_progress') if !changed_vars.empty? && formula_update_in_progress?
```

Comment:
I think we can check `formula_update_in_progress?` before we extract and calculate `changed_vars`